ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -158,19 +142,30 @@ private[backend] object H2StorageBackend
       val (wildCardClause, wildCardParams) = filterParams.wildCardParties match {
         case wildCardParties if wildCardParties.isEmpty => (Nil, Nil)
         case wildCardParties =>
+          val (clause, params) = H2QueryStrategy.arrayIntersectionNonEmptyClause(
+            columnName = witnessesColumnName,
+            parties = wildCardParties,
+            paramNamePostfix = "",
+          )
           (
-            List(s"(${arrayIntersectionWhereClause(witnessesColumnName, wildCardParties)})"),
-            Nil,
+            List(s"($clause)"),
+            params,
           )
       }
       val (partiesTemplatesClauses, partiesTemplatesParams) =
         filterParams.partiesAndTemplates.iterator.zipWithIndex
           .map { case ((parties, templateIds), index) =>
+            val (clause, params) =
+              H2QueryStrategy.arrayIntersectionNonEmptyClause(
+                witnessesColumnName,
+                parties,
+                s"ptwwc$index",
+              )
             (
-              s"( (${arrayIntersectionWhereClause(witnessesColumnName, parties)}) AND (template_id = ANY({templateIdsArraywwc$index})) )",
+              s"( ($clause) AND (template_id = ANY({templateIdsArraywwc$index})) )",
```

Comment:
nit: would it not make sense to maintain the same order for clauses and parameters?

Reply:
with the new approach this comment is obsolete